### ISSUE TITLE
Issue with dates in array of embeddables

### ISSUE THREAD

**Issue by ramiel** [mikro-orm/mikro-orm#2040]
**Describe the bug**
An embeddable have some properties which are dates and it's embedded as an array. The dates are saved as string on MongoDB instead of dates

**To Reproduce**
This is the embeddable

```ts
@Embeddable()
export class Session {
  @Property()
  id!: string;

  @Property()
  createdAt: Date = new Date();
}
```

and it's used like this

```ts
@Embedded(() => Session, { array: true, nullable: true })
sessions: Array<Session> | null = [];
```

`createdAt` is saved as `String` instead of `Date`

**Expected behavior**
createdAt should be a Date

**Versions**

| Dependency | Version |
| - | - |
| node | 14 |
| typescript | 4.2.4 |
| mikro-orm | 4.5.7 |
| your-driver | mongodb |

**Comment by EduVencovsky** [comment]
Are you sure this issue is about embeddables? If you have a date type that isn't in a embeddable, do you have the same problem?

Maybe this will help you https://mikro-orm.io/docs/custom-types/#datetype

**Comment by B4nan** [comment]
Not reproducible for me, at least not in master. Is this about what is stoerd in the database or about what gets retrieved from it (both works for me)? Please provide full repro, ideally as a test case. I tried it on [this test](https://github.com/mikro-orm/mikro-orm/blob/master/tests/features/embeddables/entities-in-embeddables.mongo.test.ts#L45), just by specifying the property initializer of already existing `IdentityLink.createdAt` property.

> Are you sure this issue is about embeddables? If you have a date type that isn't in a embeddable, do you have the same problem?

You don't need any custom type to work with `Date` objects. The one you are linking is for dates, not datetimes (which the JS `Date` object is), so when you want to store a date like `2021-12-12` instead of the full datetime, but still want to map it to `Date` object during runtime.

**Comment by ramiel** [comment]
It is about what is stored on DB, the value itself is a correct Date in JS.
If the value is not an embeddable the type on DB is Date, otherwise is a String.

**Comment by B4nan** [comment]
As said above, this is not reproducible for me, so please provide executable repro, otherwise will close this. It is possible that it got fixed in #1948 which is v5 only - I won't be backporting that to v4 so would be great if you could also verify how this behaves in v5.

**Comment by ramiel** [comment]
Look, since it's working in JS after all and it's probably fixed on v5, let's close this
